src/flow/index.js

```diff
@@ -0,0 +1,268 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { FlowChartWithState } from '@mrblenny/react-flow-chart'
+import dagre from 'dagre'
+
+import Container from './container'
+import Port from './port'
+import NodeInner from './node-inner'
+
+
+export const stepProps = Object.freeze({
+  audience_build_wi: {
+    name: 'Audience Build: Walk-in',
+    level: 0,
+  },
+  audience_build_beacon: {
+    name: 'Audience Build: Beacon',
+    level: 0,
+  },
+  audience_enrich_aoi: {
+    name: 'Enrichment: AOI',
+    level: 1,
+  },
+  audience_enrich_xd: {
+    name: 'Enrichment: Cross Device',
+    level: 1,
+  },
+  audience_intersect_vwi: {
+    name: 'Intersection: Verified Walk-in',
+    level: 2,
+  },
+  audience_intersect_xwi: {
+    name: 'Intersection: Cross Chain Walk-in',
+    level: 2,
+  },
+  segment: {
+    name: 'Segment',
+    level: 3,
+  },
+  cohort_repeat_visits: {
+    name: 'Cohort Analysis: Visits',
+    level: 3,
+  },
+  cohort_converted_visitors: {
+    name: 'Cohort Analysis: Converted Visitors',
+    level: 3,
+  },
+  propensity: {
+    name: 'Propensity',
+    level: 3,
+  },
+  report_wi: {
+    name: 'Report: Walk-in',
+    level: 3,
+  },
+  report_vwi: {
+    name: 'Report: Verified Walk-in',
+    level: 3,
+  },
+  report_xwi: {
+    name: 'Report: Cross Chain Walk-in',
+    level: 3,
+  },
+})
+
+export const transform = ({ job_parameters, dag_tasks = [], scale = 1 }) => {
+  const data = {
+    offset: { x: 0, y: 0 },
+    scale,
+    selected: {},
+    hovered: {},
+  }
+  const { steps = [] } = job_parameters || {}
+
+  const nodes = []
+  const links = []
+  // stateful helper functions
+  function buildLink(target, src) {
```

Comment:
this looks similar to the buildLink function in dagre/index.js except what object is being pushed is different. Maybe consider turning this into a helper function which takes in an argument (in flow/index.js vs dagre/index.js file).